js/app.js

```diff
@@ -219,28 +256,67 @@ function pathIntoTitleInput(paste) {
 }
 
 function storeTags() {
+  const currentDomain = document.getElementById("domain-tag").innerHTML;
   const url = linkInput.value;
   const title = titleInput.value;
   let newTags = document.getElementsByClassName("tagged");
   let existTags = document.getElementsByClassName("exist-tagged");
-  //let largerArray = Math.max(newTags.length, existTags.length);
   let tileNo = Object.keys(STATE.tiles).length + 1;
   let newTile = "tile" + tileNo;
+  let domainNo = Object.keys(STATE.domains).length + 1;
+  let newDomain = "domain" + domainNo;
 
-  STATE.tiles[newTile] = { id: tileNo, url: url, title: title, tags: [] };
+  // new tile object
+  STATE.tiles[newTile] = {
+    id: tileNo,
+    domain: currentDomain,
+    url: url,
+    title: title,
+    tags: [],
+  };
 
+  // adds new tags to tags object
+  // and adds new tags to new tile object
   for (var i = 0; i < newTags.length; i++) {
     newTagProp = newTags[i].textContent;
     STATE.tags[newTagProp] = { tiles: [] };
     STATE.tags[newTagProp].tiles.push(newTile);
     STATE.tiles[newTile].tags.push(newTagProp);
   }
 
+  // adds new tile to existing tag object
+  // and adds tag to new tile object
   for (var i = 0; i < existTags.length; i++) {
     existTagProp = existTags[i].textContent;
     STATE.tags[existTagProp].tiles.push(newTile);
     STATE.tiles[newTile].tags.push(existTagProp);
   }
+
+  domainArray = [];
+  /**
+   * Loops through domain objects
+   * If not the same as URL domain, then puts it into an array
+   * If it is, then it pushes the tile reference to the existing domain object
+   * Also puts it into the array so that new domain object constructor only occurs when it's a new domain URL
+   */
+  for (const existDomain of Object.keys(STATE.domains)) {
+    if (STATE.domains[existDomain].domain != currentDomain) {
+      domainArray.push(STATE.domains[existDomain].domain);
+    } else if (STATE.domains[existDomain].domain === currentDomain) {
+      STATE.domains[existDomain].tiles.push(newTile);
+      domainArray.push(currentDomain);
+    }
+  }
+
```

Comment:
Now with the new STATE.domain approach you can simplify this pretty much with something like ` if (domains[newDomain])`